Vec2 startPoint
Vec2 startVelocity
Vec2 endPoint
Vec2 endVelocity